### PR TITLE
Object flatten

### PR DESCRIPTION
**Pull request by satazor**

### CODE REVIEW

**Comment by satazor** [comment]
@jdalton this is similar to https://www.npmjs.com/package/flat

**Comment by jdalton** [comment]
>  this is similar to https://www.npmjs.com/package/flat

The `unflatten` in `flat` is hawt.

**Comment by satazor** [comment]
Yeah, but if we had it we would be missing unflatten for arrays becaus its impossible to reverse the operation for them :(

File: src/object/flatten.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            return {};
+        }
+
+        level = level == null ? -1 : level;
+        return flattenTo(obj, {}, '', level);
+    }
+
+    return flatten;
+
+});
```
Comment (millermedeiros) [inline src/object/flatten.js]: missing line break at end of file

**Comment by millermedeiros** [comment]
this is pretty cool. I liked the behavior described on the tests (not messing with arrays). Maybe we should mention `flat` in the docs as an alternative.

we could implement `unflatten` for objects in the future if needed.

File: src/object/flatten.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+define(['../lang/isPlainObject', './mixIn'], function (isPlainObject, mixIn) {
```
Comment (millermedeiros) [inline src/object/flatten.js]: `mixIn` is not really being used.

File: src/object/flatten.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    /*
+     * Helper function to flatten to a destination object.
+     * Used to remove the need to create intermediate objects while flattening.
+     */
+    function flattenTo(obj, result, prefix, level) {
+        var key,
+            value,
+            nestedPrefix;
+        for (key in obj) {
```
Comment (millermedeiros) [inline src/object/flatten.js]: what about using `forOwn` here? just in case user augmented the protype..

**Comment by satazor** [comment]
@millermedeiros applied the changes based on your feedback.

**Comment by millermedeiros** [comment]
LGTM :+1:

**Comment by satazor** [comment]
Can I merge it?

**Comment by millermedeiros** [comment]
@satazor sure!

**Pull request merged**